homeassistant/components/mqtt/strings.json

```diff
@@ -3,6 +3,28 @@
     "invalid_platform_config": {
       "title": "Invalid config found for MQTT {domain} item",
       "description": "Home Assistant detected an invalid config for a manually configured item.\n\nPlatform domain: **{domain}**\nConfiguration file: **{config_file}**\nNear line: **{line}**\nConfiguration found:\n```yaml\n{config}\n```\nError: **{error}**.\n\nMake sure the configuration is valid and [reload](/developer-tools/yaml) the manually configured MQTT items or restart Home Assistant to fix this issue."
+    },
+    "subentry_migration_discovery": {
+      "title": "MQTT device \"{name}\" subentry migration to MQTT discovery",
+      "fix_flow": {
+        "step": {
+          "confirm": {
+            "title": "[%key:component::mqtt::issues::subentry_migration_discovery::title%]",
+            "description": "Exported MQTT device \"{name}\" identified via MQTT discovery. Select **Submit** to confirm that the MQTT device is to be migrated to the main MQTT configuration, and to remove the existing MQTT device subentry. Make sure that the discovery is retained at the MQTT broker, or is resent after the subentry is removed, so that the MQTT device will be set up correctly."
```

Comment:
I think the repair should explain that the other alternative is to change the device identifier (same for the yaml case).

Reply:
Added. The explanation was already in the export dialogs, but it does no harm to explain this can still be done.

Reply:
Updated the screenshots in the PR description